src/main/java/appeng/core/Registration.java

```diff
@@ -338,9 +327,9 @@ void postInit( final FMLPostInitializationEvent event )
 	{
 		this.registerSpatial( true );
 
-		final IAppEngApi api = AEApi.instance();
+		final Api api = Api.INSTANCE;
```

Comment:
> In this case I need access to the Factory, but I am not 100% happy about the flow in which the API is instantiated (static field). Would it be possible to use a different mechanism of exposing the API to mods? Or at the very least only expose it once AE begins it's own initailization? 

No idea currently. The API might still be an artifact from the beginnings of 1.7.10 and forge missing many features. Iirc forge can now inject instanciated mods into other mods. But I have no idea how reliable it is. Or maybe look at JEI which also injects the API into any mod using the correct annotation instead of having a mod trying to access it directly.

> What is the reasoning for exporting all blocks/items in the way they are currently exposed in the API. Wouldn't it be sufficient to just expose constants for our registry names?

The API provides a bit more as just the items. Also helper methods to compare them regarding different `AEColors`. And addons can use them with `API.stuff.item().ifPresent(is -> doStuffWith(is))` and not query the gameregistry for it, use null checks and what not.